### ISSUE TITLE
[Accessibility]: Search contexts management ARC checklist issues

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34741]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34191

### Problem description

The following ARC toolkit issues exist in the [search context management pages](https://sourcegraph.com/contexts): 

- [ ] Headers should not be H1, should be H2 styled as H1, on all pages ([list page](https://sourcegraph.com/contexts), [view page](https://sourcegraph.com/contexts/global), and [add/edit page](https://sourcegraph.com/contexts/new))
![CleanShot 2022-04-29 at 16 47 04](https://user-images.githubusercontent.com/206864/166081367-c4d85f36-e771-4f00-a17d-d77b0fa905fd.png)
![CleanShot 2022-04-29 at 16 47 22](https://user-images.githubusercontent.com/206864/166081377-06ac719f-8fc5-4c8b-929c-30dc1c01ecf2.png)
![image](https://user-images.githubusercontent.com/206864/166081393-280eb22c-3128-43fa-be80-06c425718e5c.png)
- [ ] On list page, `<ul>` should use `<li>` instead of `<div>` for children
![CleanShot 2022-04-29 at 16 46 38](https://user-images.githubusercontent.com/206864/166081352-2fd53b4b-e6c8-4927-a971-38e8a9c80bb3.png)
- [ ] On list page, "Filter search context" textbox should have an aria-label
![image](https://user-images.githubusercontent.com/206864/166081406-659d4678-be9f-4f9a-8c3e-422d1443395e.png)
- [ ] On edit page, Delete button doesn't meet color contrast requirements (should use a danger variant button instead)
![CleanShot 2022-04-29 at 16 48 21](https://user-images.githubusercontent.com/206864/166081428-45e08eed-ab68-4739-baf8-06d900c2601f.png)

### Expected behavior

All issues should be fixed as noted above

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by gitstart-sourcegraph** [comment]
## Related Issue
https://github.com/sourcegraph/sourcegraph/issues/35613
